### PR TITLE
FOR-224: Added signup form

### PR DESCRIPTION
**Pull request by tuj**
#### Ticket

https://jira.itkdev.dk/browse/FOR-224

#### Description

- Adds a sign up form for parents.
- Adds d9 config changes that was omitted from the original d9 release.
- Changes Readme
- Add honeypot and symfony mailer modules and related config
- Add template for system messages.
- Add template for signup mail.
- Add /sites/default/settings.local.php to .gitignore

#### Screenshot
screenshot-foreningsmentor local itkdev dk-2022 04 01-09_35_31
screenshot-foreningsmentor local itkdev dk-2022 04 01-09_36_00

### CODE REVIEW

File: web/modules/custom/foreningsmentor/foreningsmentor.module [hunk 1]
```diff
@@ -423,3 +407,26 @@ function foreningsmentor_views_pre_render(ViewExecutable $view) {
...
+ * @throws \Twig\Error\RuntimeError
+ */
+function foreningsmentor_mail($key, &$message, $params) {
+  if ('signup' === $key) {
+    $message['headers']['Content-Type'] = 'text/html; charset=UTF-8; format=flowed; delsp=yes';
+    $mailContent = twig_render_template(\Drupal::service('extension.list.module')->getPath('foreningsmentor') . '/templates/signup-email-body.html.twig', array(
+      'params' => $message['params'],
+      // Needed to prevent notices when Twig debugging is enabled.
+      'theme_hook_original' => 'not-applicable',
+    ));
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/foreningsmentor.module]: Use short array syntax (`[…]`) for consistency with the rest of the code.

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<p>{{ params.form_values.child_name }}</p>
+<h4>Barnets fødselsdag</h4>
+<p>{{ params.form_values.birth_date }}</p>
+<h4>Barnets køn</h4>
+<p>{{ params.form_values.sex }}</p>
+<h4>Barnets skole</h4>
+<p>{{ params.form_values.school }}</p>
+<h4>Barnets klassetrin</h4>
+<p>{{ params.form_values.grade }}</p>
+<h2>Forældre information</h2>
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: Should be either “Forælders information” (for consistency with “Barnets information”) or “Forælderinformation”.

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<h1>Ny tilmelding</h1>
+<p>Der er kommet en ny tilmelding på {{ params.site_name }}.</p>
+<h2>Barnets information</h2>
+<h4>Barnets navn</h4>
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: The next header level after 2 is 3, i.e. `<h3>` should be used here.

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+<p>{{ params.form_values.child_name }}</p>
+<h4>Barnets fødselsdag</h4>
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: “Barnets ” could be removed since the heading already tells that this is information about the child.

File: web/sites/default/settings.php [hunk 5]
```diff
@@ -762,18 +762,6 @@
...
-  'username' => getenv('DATABASE_USERNAME') ?: 'db',
-  'password' => getenv('DATABASE_PASSWORD') ?: 'db',
-  'host' => getenv('DATABASE_HOST') ?: 'mariadb',
-  'port' => getenv('DATABASE_PORT') ?: '',
-  'driver' => getenv('DATABASE_DRIVER') ?: 'mysql',
-  'prefix' => '',
-];
-
-$settings['config_sync_directory'] = '../config/sync';
-
```
Comment (rimi-itk) [inline web/sites/default/settings.php:776]: Why is this removed?

File: .gitignore [hunk 6]
```diff
@@ -184,7 +184,7 @@
 # ------------------------------------------------------------------------------
 
 # We want all custom site settings in settings.local.php
-!/web/sites/*/settings*.php
+/web/sites/*/settings*.php
```
Comment (rimi-itk) [inline .gitignore]: Shouldn't this be `/web/sites/*/settings.local.php` to ignore custom site settings and keep `web/sites/*/settings.php` under version control?

File: README.md [hunk 7]
```diff
@@ -1,87 +1,69 @@
...
+ 'database' => getenv('DATABASE_DATABASE') ?: 'db',
+ 'username' => getenv('DATABASE_USERNAME') ?: 'db',
+ 'password' => getenv('DATABASE_PASSWORD') ?: 'db',
+ 'host' => getenv('DATABASE_HOST') ?: 'mariadb',
+ 'port' => getenv('DATABASE_PORT') ?: '',
+ 'driver' => getenv('DATABASE_DRIVER') ?: 'mysql',
+ 'prefix' => '',
+];
+
+$settings['config_sync_directory'] = '../config/sync';
```
Comment (rimi-itk) [inline README.md:27]: I think this should bet set in `settings.php` (too).

File: web/modules/custom/foreningsmentor/foreningsmentor.module [hunk 8]
```diff
@@ -423,3 +407,26 @@ function foreningsmentor_views_pre_render(ViewExecutable $view) {
...
+ *
+ * Change mail header and body.
+ *
+ * @param $key
+ * @param $message
+ * @param $params
+ * @throws \Twig\Error\RuntimeError
+ */
+function foreningsmentor_mail($key, &$message, $params) {
+  if ('signup' === $key) {
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/foreningsmentor.module]: Are we hooking into a generic Drupal signup processes here? Otherwise, I think a more specific mail key, `foreningsmentor_signup`, say, should be used.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some minor issues and comments should be addressed.

File: web/modules/custom/foreningsmentor/foreningsmentor.module (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/foreningsmentor.module]: Changed

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig (hunk 2 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: Changed

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig (hunk 3 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: Fixed

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig (hunk 4 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: Changed

File: web/sites/default/settings.php (hunk 5 above)
Comment (martinyde) [reply web/sites/default/settings.php:776]: Added

File: .gitignore (hunk 6 above)
Comment (martinyde) [reply .gitignore]: Fixed

File: README.md (hunk 7 above)
Comment (martinyde) [reply README.md:27]: Added

File: web/modules/custom/foreningsmentor/foreningsmentor.module (hunk 8 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/foreningsmentor.module]: Changed

**Review by martinyde — APPROVED** [review]

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig [hunk 9]
```diff
@@ -1,26 +1,26 @@
...
 <p>{{ params.form_values.sex }}</p>
-<h4>Barnets skole</h4>
+<h3>Skole</h3>
 <p>{{ params.form_values.school }}</p>
-<h4>Barnets klassetrin</h4>
+<h3>Klassetrin</h3>
 <p>{{ params.form_values.grade }}</p>
-<h2>Forældre information</h2>
-<h4>Navn</h4>
+<h2>Forældreinformation</h2>
```
Comment (rimi-itk) [inline web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: This should be “Forælderinformation” (there's only one parent in this case), cf. https://dsn.dk/ordbog/ro/foraelder/.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig (hunk 9 above)
Comment (martinyde) [reply web/modules/custom/foreningsmentor/templates/signup-email-body.html.twig]: Changed

**Review by rimi-itk — APPROVED** [review]
